pyproject.toml

```
@@ -145,8 +117,14 @@ ignore = [
     "Q000", # 2 single-quoted strings - probably accidental
     "RET504", "RET506",  # Return value related.
     "PLR2004", # Magic value used in comparison
+    "S603", # `subprocess` call: check for execution of untrusted input
+    "S607", # Starting a process with a partial executable path
 
 ]
+[tool.ruff.pylint]
+max-args = 24
```

Let's keep this the same right now

```suggestion
max-args = 6
```

---

OK, then I am going to add all the noqa to the affected functions

---

Thank you!

I'd rather have noqa comments for things like this since they help make code smells more visible. There's some ongoing discussion about bumping it up further at https://github.com/canonical/starbase/issues/133

---

24 is a bit extreme but craft-parts has many calls with more than six arguments, not sure if it's productive to leave it at a value where we'll need too many PLR0913s.